scene/2d/polygon_2d.cpp

```diff
@@ -302,17 +302,19 @@ void Polygon2D::_notification(int p_what) {
 					colors.write[i] = color_r[i];
 				}
 			} else {
+				colors.resize(len);
+				for (int i = 0; i < len; i++) {
+					colors.write[i] = color;
+				}
 				colors.push_back(color);
```

Comment:
not sure why +1, but it is intended, as I had to change Polygon2D to use meshes in order to work again with skeletons.

Reply:
+1 because you `resize(len)` and then still `push_back(color)`. So you get a size of `len + 1`.
Looks to me like you wanted to replace the `push_back` with the resize + `for`?